Tests/BasicsTests/SandboxTests.swift

```diff
@@ -86,72 +100,106 @@ final class SandboxTest: XCTestCase {
         let tmpFile1 = NSTemporaryDirectory() + "/" + UUID().uuidString
-        let command1 = Sandbox.apply(command: ["touch", tmpFile1], strictness: .writableTemporaryDirectory)
+        let command1 = SandboxProfile(allowWritingToTemporaryDirectories: true).apply(to:
+            ["touch", tmpFile1])
         XCTAssertNoThrow(try Process.checkNonZeroExit(arguments: command1))
         try? FileManager.default.removeItem(atPath: tmpFile1)
 
         let tmpFile2 = "/tmp" + "/" + UUID().uuidString
-        let command2 = Sandbox.apply(command: ["touch", tmpFile2], strictness: .writableTemporaryDirectory)
+        let command2 = SandboxProfile(allowWritingToTemporaryDirectories: true).apply(to:
+            ["touch", tmpFile2])
         XCTAssertNoThrow(try Process.checkNonZeroExit(arguments: command2))
         try? FileManager.default.removeItem(atPath: tmpFile2)
     }
 
     func testWritingToReadOnlyInsideWritableNotAllowed() throws {
         #if !os(macOS)
-        try XCTSkipIf(true, "test is only supported on macOS")
+        try XCTSkipIf(true, "sandboxing is only supported on macOS")
         #endif
 
         try withTemporaryDirectory { tmpDir in
-            // Check that we can write into it, but not to a read-only directory underneath it.
+            // Check that we can write into it, but not into a read-only directory underneath it.
             let writableDir = tmpDir.appending(component: "ShouldBeWritable")
-            let allowedCommand = Sandbox.apply(command: ["mkdir", writableDir.pathString], strictness: .default, writableDirectories: [tmpDir])
+            try localFileSystem.createDirectory(writableDir)
+            let allowedCommand = SandboxProfile(pathRules: [.writable(writableDir)]).apply(to:
+                ["touch", writableDir.pathString])
             XCTAssertNoThrow(try Process.checkNonZeroExit(arguments: allowedCommand))
 
             // Check that we cannot write into a read-only directory inside it.
             let readOnlyDir = writableDir.appending(component: "ShouldBeReadOnly")
             try localFileSystem.createDirectory(readOnlyDir)
-            let deniedCommand = Sandbox.apply(command: ["touch", readOnlyDir.pathString], strictness: .default, writableDirectories: [tmpDir], readOnlyDirectories: [readOnlyDir])
+            let deniedCommand = SandboxProfile(pathRules: [.writable(tmpDir), .readonly(readOnlyDir)]).apply(to:
+                ["touch", readOnlyDir.pathString])
+            XCTAssertThrowsError(try Process.checkNonZeroExit(arguments: deniedCommand)) { error in
+                guard case ProcessResult.Error.nonZeroExit(let result) = error else {
+                    return XCTFail("invalid error \(error)")
+                }
+                XCTAssertMatch(try! result.utf8stderrOutput(), .contains("Operation not permitted"))
+            }
+        }
+    }
+
+    func testWritingToWritableInsideReadOnlyAllowed() throws {
```

Comment:
This adds a specific test case for a writable directory inside of a read-only directory.